packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -769,7 +770,10 @@ export class ResourceService {
                 if (!fieldProperties.allowMultipleSelection) {
                   createFieldInput.name = `${createFieldInput.name}Id`;
                   createFieldInput.displayName = `${createFieldInput.displayName} ID`;
-                  createFieldInput.dataType = EnumDataType.SingleLineText; //@todo: set the new type based on the ID type of the related entity (WholeNumber or SingleLineText)
+                  createFieldInput.dataType =
+                    await this.entityService.updateFieldDataTypeIdByRelatedEntity(
+                      relatedEntityId
+                    );
                   const fieldProperties: SingleLineText = {
```

Comment:
I think the field properties are no longer relevant since they were hard-coded for SIngleLineText field props